### ISSUE TITLE
Electron 9.x not closing WS connections

### ISSUE THREAD

**Issue by peteringram0** [electron/electron#24389]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 9.0.5
* **Operating System:**
  * macOS 10.15.5
* **Last Known Working Electron version:**
  * 8.3.4

### Expected Behavior
Websockets within Iframes seem to get held open in Electron 9 where as electron 8 when removed the connections get closed correctly.

### Actual Behavior
Websockets should get closed as they do in 8.3.4

### Screenshots
Host an iframe that holds a Websocket connection which opens on load and runs .close() beforeunload. You will see the beforeUnload event gets fired correctly but the connection stays open.

**My application running Electron 8.3.4**
<img width="1059" alt="Screenshot 2020-07-01 at 17 05 17" src="https://user-images.githubusercontent.com/6588942/86267538-c3b7bb80-bbbe-11ea-8e24-7f3e6205db84.png">

**My application running Electron 9.0.5**
<img width="1055" alt="Screenshot 2020-07-01 at 17 14 16" src="https://user-images.githubusercontent.com/6588942/86267592-d9c57c00-bbbe-11ea-8e42-c173c486ff07.png">

On the images above the only change on the application is the electron version and nothing else. As you can see the connections on 9.x stay in 'Pending' state and never get closed which is different from 8.x.

**Comment by nornagon** [comment]
It would be helpful to have a small repro case for this, for example using [Electron Fiddle](/electron/fiddle).

**Comment by peteringram0** [comment]
@nornagon 

Here you go: https://gist.github.com/peteringram0/5355f2183caadd52c602fac541c36635

I have tested with Electron Fiddle and getting the same results as above with this minimal repo. Please can you un-block. Thanks.

**Comment by peteringram0** [comment]
@nornagon can this be unblocked please? I think this is a pretty big issue on Electron 9.0.0 >

**Comment by nornagon** [comment]
@peteringram0 thanks for the repro, but the part with the websocket code doesn't seem to be part of the fiddle (it seems to load an iframe). That makes it difficult to determine what's actually happening with the websocket itself.

I do observe in the network tab of devtools that the WS connections appear to remain 'pending', though.

**Comment by peteringram0** [comment]
Hi @nornagon  Yer thats kinda the point, that there is an issue with Electron 9.x when iframe open / close a websocket. If you switch to Electron 8 you will notice its working fine.

The page in question only really opens a websocket on load and closes it before destroy.

````javascript

// Runs when page loads
const socket = new WebSocket("websocket path");

// Runs with page unloads
socket.close(1000)

````

**Comment by peteringram0** [comment]
Is anyone else seeing this issue?

**Comment by peteringram0** [comment]
Still not fixed on v9.1.1 incase anyone is wondering.

**Comment by peteringram0** [comment]
Tested on v9.1.2, still has issues.

**Comment by peteringram0** [comment]
Tested on v9.2.1 still no fix.

**Comment by nornagon** [comment]
Appears to be fixed in 10.x.

**Comment by nornagon** [comment]
Closing as this is fixed in 10.x, however if it's determined what the issue was and that the fix is backportable to 9.x, I'd consider backporting it.

**Comment by peteringram0** [comment]
You'r right it seems to be fixed even in 10.0.0-Beta.1. Im guessing this could be a fix inside chrome?

**Comment by nornagon** [comment]
Yeah, that seems plausible.
